### PR TITLE
Austenem/refactor ingest metadata

### PR DESCRIPTION
**Pull request by austenem**
- Adjusts portal translation to account for refactoring of `metadata` fields in the entity-api.

  The relevant entity-api changes include:

    `Dataset.ingest_metadata.metadata` → `Dataset.metadata`
    `Dataset.ingest_metadata.files` → `Dataset.files`
    `Dataset.metadata` → `Dataset.ingest_metadata` (Elasticsearch/Search API only)

### CODE REVIEW

File: src/hubmap_translation/addl_index_transformations/portal/.flake8
Comment (austenem) [inline src/hubmap_translation/addl_index_transformations/portal/.flake8:1]: I was unable to install the flake8 dependency without this change - I'm assuming this was not an issue in past installations.

File: src/hubmap_translation/addl_index_transformations/portal/translate.py [hunk 1]
```diff
@@ -44,30 +43,6 @@ def _map(doc, key, map):
             _map(ancestor, key, map)
 
 
-def _add_metadata_metadata_placeholder(doc):
```
Comment (john-conroy) [inline src/hubmap_translation/addl_index_transformations/portal/translate.py:47]: I think some folks still use the dev-search feature. You'll have to make an accompanying change to handle https://github.com/hubmapconsortium/portal-ui/blob/0cb4529dcf06ccfb05317f7ed9891291be002a77/context/app/static/js/pages/search/DevSearch.jsx#L87.

**Review by john-conroy — APPROVED** [review]
Looks good to me. The plan is to get this out on dev and then work on the changes to `portal-ui`?

File: src/hubmap_translation/addl_index_transformations/portal/translate.py (hunk 1 above)
Comment (austenem) [reply src/hubmap_translation/addl_index_transformations/portal/translate.py:47]: Got it!

**Comment by austenem** [comment]
> Looks good to me. The plan is to get this out on dev and then work on the changes to `portal-ui`?

Thanks for the review, that's correct.

File: src/hubmap_translation/addl_index_transformations/portal/translate.py (hunk 1 above)
Comment (yuanzhou) [reply src/hubmap_translation/addl_index_transformations/portal/translate.py:47]: Good point @john-conroy.  Personally I use that `/dev-search` endpoint for reindex verification.

**Comment by yuanzhou** [comment]
@austenem originally I thought you would branch off `karlburke/metadataReorg` and apply the portal-specific updates, otherwise it would be hard for you to test locally without those changes. As long as @lchoy @john-conroy @NickAkhmetov approve your changes to the portal pieces, I'm fine with moving forward with your current branch. We'll just have to get the changes from `karlburke/metadataReorg` and your branch merged/deployed altogether when moving to TEST and PROD.

**Comment by lchoy** [comment]
> @austenem originally I thought you would branch off `karlburke/metadataReorg` and apply the portal-specific updates, otherwise it would be hard for you to test locally without those changes. As long as @lchoy @john-conroy @NickAkhmetov approve your changes to the portal pieces, I'm fine with moving forward with your current branch. We'll just have to get the changes from `karlburke/metadataReorg` and your branch merged/deployed altogether when moving to TEST and PROD.

@yuanzhou The desired base branch may not have been communicated to @austenem. I pulled this branch locally, rebased to `karlburke/metadataReorg`, and successfully ran automated and manual tests.

**Review by yuanzhou — APPROVED** [review]

**Pull request merged**